raco/backends/myria/myria.py

```diff
@@ -1721,6 +1824,193 @@ def fire(self, op):
         return algebra.UnionAll(children)
 
 
+class FillInJoinPullOrder(rules.Rule):
+
+    def __init__(self, pull_order):
+        self._disabled = False
+        self.pull_order = pull_order
+
+    def fire(self, op):
+        if not isinstance(op, MyriaSymmetricHashJoin):
+            return op
+        op.pull_order = self.pull_order
+        return op
+
+
+class PropagateAsyncFTBuffer(rules.Rule):
+
+    def __init__(self):
+        self._disabled = False
+
+    def fire(self, op):
+        if not isinstance(op, MyriaIDBController):
+            return op
+        buffer_type = op.get_agg_json()
+        if buffer_type['type'] == "CountFilter":
+            # do not propagate CountFilter
+            return op
+        # only do it for the iterative input
+        if isinstance(op.args[1], MyriaShuffleConsumer):
+            op.args[1].input.buffer_type = buffer_type
```

Comment:
I didn't add `buffer_type` as one arg of `__init__` since this information is not available until `PropagateAsyncFTBuffer`, but `MyriaShuffleProducer`s are constructed before this rule. I added a set method to set the buffer type.